### DISCUSSION TITLE
How to turn an instance of Cats `IO` to `ConnectionIO`

### DISCUSSION THREAD

**Discussion by stefanfromm-allegro** [tpolecat/doobie#1744] [Q&A] [answered]
I'm using doobie 1.0.0-RC2 and Cats Effect 3 and would like to turn an instance of Cats `IO` to `ConnectionIO`. In earlier versions this was possible by

```scala
Async[ConnectionIO].liftIO(catsIO)
```

But an instance of `Async[ConnectionIO]` has gone. In [the meanwhile](https://github.com/tpolecat/doobie/commit/3b387a4ae5209682433b14c1044c891e094cfd02) there was an instance of `LiftIO[ConnectionIO]`, but it is also gone. https://stackoverflow.com/a/50481203 states that the following approach worked:

```scala
catsIO.to[ConnectionIO]
```

but I receive the error "No given instance of type cats.effect.LiftIO[doobie.ConnectionIO] was found for parameter F of method to in class IO". I cannot find an instance of `LiftIO`.

Does anyone know of how to turn an instance of Cats `IO` to `ConnectionIO`?

**Comment by jatcwang** [comment, marked as answer]
For CE3/Doobie 1.x: https://stackoverflow.com/a/71257623

**Reply by i-am-the-slime** [reply]
I fail to understand this.
It's nice, that I can get the current time with `Clock`, but I can't mock this time in my tests.

**Reply by jatcwang** [reply]
@i-am-the-slime If you want to have control over time in your tests, https://typelevel.org/cats-effect/docs/core/test-runtime might help. I don't think there's anything in Doobie that stops you from using the CE TestRuntime?

**Reply by i-am-the-slime** [reply]
I was getting an error that something uses `Async` and that I should rewrite my code in order to be able to test it.

**Reply by jatcwang** [reply]
If you can post a small minimized example I should be able to point you towards the right direction

**Reply by i-am-the-slime** [reply]
I could work around it by just using `Sync`. I amended the stack overflow answer.
